### ISSUE TITLE
Oct 1 - pc searches and mobile searches are not running

### ISSUE THREAD

**Issue by specu** [sealemar/BingRewards#279]
let it begin....

**Comment by DARKKRAKEN** [comment]
Yeah seems to be broken again.

**Comment by coollbv** [comment]
Same...

**Comment by FadeVT** [comment]
Same here.

**Comment by baltimoron** [comment]
It looks like https://www.bing.com/rewardsapp/flyoutpage?style=v2 is only returning the coffee break quiz RE (if you turn on verbose mode and run the script you can see this). Might need to switch the page that is parsed for the reward REs (could use the dashboard) but that would require a total rewrite of bingFlyoutParser.py. I'll look into that but for now I think I'm just gonna hard-code the number of searches.

**Comment by duddles** [comment]
@baltimoron Can you explain where/how to hard-code the # of searches? Thanks

**Comment by DimaRazb** [comment]
Same and it began 29 on me.

escription : Earn 10 poin
  Message  :
  Action   : **pass**

esult 11/11:
----------
ame        : Your goal
rogressCur : 30540
rogressMax : 24500
rogress %  : 124.65%
escription : 
  Message  :
  Action   : **inform**

**Comment by baltimoron** [comment]
@duddles try the attached version of bingRewards.py. I run a very custom version of this script so I had to copy all the changes over to the stock version. I might have missed a line so let me know if it doesn't work.

**Comment by baltimoron** [comment]
[bingRewards.zip](https://github.com/sealemar/BingRewards/files/1347515/bingRewards.zip)
@duddles try the attached version of bingRewards.py. I run a very custom version of this script so I had to copy all the changes over to the stock version. I might have missed a line so let me know if it doesn't work.

**Comment by DARKKRAKEN** [comment]
@baltimoron Seems to work great. I'm guessing this is only a stop gap fix until it can be fixed properly?

**Comment by baltimoron** [comment]
@DARKKRAKEN yeah this is definitely just a stopgap. Rewriting the parser might take a while.

**Comment by gulliverbear** [comment]
@baltimoron Thanks your file is working for me, I appreciate it

**Comment by duddles** [comment]
@baltimoron Thanks, your file is working for me

**Comment by baltimoron** [comment]
Should note that this doesn't check to see if any searches have been run for the account yet. So even if your searches for the day are complete, running the script again will cause all the searches to run again. Also doesn't check for the daily 10 point tasks (same issue with the regex) so you'll only get search points.

**Comment by someuserx** [comment]
I thought the script only worked if you had reached level 2 for the month. Unless this is a different issue?

**Comment by coollbv** [comment]
This is a different issue. The bingrewards.py that @baltimoron provided works.

**Comment by FadeVT** [comment]
@baltimoron Thank you very much for posting that zip file! It took care of the issue for me beautifully!

**Comment by adambird79** [comment]
I get for a couple of my accounts the following:

Live - email@hotmail.com
for:  Live - email@hotmail.com

No searches are made on the account but the updated bingrewards.py does work for 2 of my 4 accounts.

**Comment by kayla210** [comment]
@baltimoron Thanks so much for sharing your changes!

**Comment by imckumar** [comment]
@baltimoron Thanks for Sharing. Works perfectly.

**Comment by kenneyhe** [comment]
@baltimoron  looks good but handling "blank reward" can be problematic on tuesday with double points since you hard coded to 30...

@adambird7  my fix in mp branch (https://github.com/kenneyhe/BingRewards) pkg/bingReward.py

**Comment by adambird79** [comment]
@kenneyhe when I copy the bingrewards.py I get:

Live - email@hotmail.com
ValueError: handling substring not found

for: Live - email@hotmail.com

I have checked and double checked my config and even tried creating a new config with only 1 account. 

Thanks

**Comment by specu** [comment]
@kenneyhe why did you choose to create a fork in the first place?

**Comment by kenneyhe** [comment]
@adambird79 handling substring not found just means that some values are not in order expected. sometimes it means browser search, mobile and PC search are not found in that respective order.  I don't understand your issue. Is the fix working for PC search/mobile search?  There are lots of issue with the bing's flyout page. a rewrite to use bing dashboard is best.   My code verify with dashboard result

@specu every one creates a fork for their fixes and workaround so we don't have to touch the upstream master branch.  I have multiprocess that speeds up my bingreward for example.

**Comment by DarkAlexWang** [comment]
Can anyone do a PR instead of sharing zip files?

**Comment by elisiano** [comment]
@DarkAlexWang I've tried in the past to bring this up, I was ignored 😅

**Comment by DarkAlexWang** [comment]
@elisiano I saw your post and I am 100% supporting it. I guess we really need someone to constantly organize this repo.

**Comment by specu** [comment]
I think everyone has as much time as anyone else. I dont think that zip file needs to be put into PR. I think the worksaround is much simpler: logic would be if no PC searches are found do 30 searches + random extras and same for 20 mobile searches. That would still be a workaround not a fix. I'm planning to do that, but when i have time. if someone beats me to it, +1 to them.
Long term though i think the URLs are taken from bing rewards app, so i guess app was updated and uses new URLs - those need to be collected and new parser needs to be written. Or just parse what PC browser sees.

**Comment by baltimoron** [comment]
@DarkAlexWang The file I provided is a temporary quick fix while I work on a better solution so I don't think it really needs a pull request. I'm going to rewrite the parser to use the dashboard when I have some time and will make that one a PR.

**Comment by kenneyhe** [comment]
@DARKKRAKEN @elisiano I agree. I have my unittests and fork branch that is ready but I do not have permission for master branch upstream.

**Comment by specu** [comment]
@kenneyhe many of us dont have permission, but we all can create a pull request and an admin would review and merge if it looks good.

**Comment by kenneyhe** [comment]
@specu many of us do not have this problem.  Branches are very upstream and unittesting. The code quality and coverage can be improved with a CD badge like circleci/codeship badge.

**Comment by specu** [comment]
it's all over my head. I'm saying you dont need master branch upstream, just create a PR for your changes. someone would review and merge if they are good. there's no problem.

**Comment by kenneyhe** [comment]
@specu I don't think we all agree on the PR merging scheme. PR to master is not agreed upon. Hence, PR can not be created because we actually want to merge to a branch. Problem can be solved by someone who can create branches so every patch to that branch is simpler to merge to master.

**Comment by specu** [comment]
i'm sure if you prove yourself to trustworthy with those privileges, the admins would have no problem giving you that access.

**Comment by kenneyhe** [comment]
@specu not my call. I have a day job so I am happy working on my own branches

**Comment by amayer5125** [comment]
@kenneyhe If you make a change and would like it merged I would be more than happy to review it. Just make a merge request and I will test it. I am not saying I will test it the same day as, like you, I also have a full time job. But I try my best to say on top of Pull Requests and test them.

**Comment by kenneyhe** [comment]
PR #283 is ready but all you really need is a patch since my improvement for everything else is being differ by github PR ui

**Comment by baltimoron** [comment]
I rewrote the parser to use the dashboard but can't figure out the pull request process (never used git outside of this project). Based on what I'm seeing I need to create a branch then compare to master but it doesn't look like I have to ability to create a branch. Can anyone fill me in here?

**Comment by specu** [comment]
@baltimoron no, you need to create a fork of https://github.com/sealemar/BingRewards
then apply your changes to your fork and then on the main page of your fork of the repository you got a button "new pull request" - please let me know if that worked (maybe that should be part of the documentation for this project @amayer5125 )

**Comment by baltimoron** [comment]
@specu thanks, looks like it worked. https://github.com/sealemar/BingRewards/pull/284

**Comment by DARKKRAKEN** [comment]
Thanks for the fix baltimoron!
